drivers/display/ek79652.c

```diff
@@ -0,0 +1,552 @@
+/*
+ * Copyright (c) 2021 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gooddisplay_ek79652
+
+#include <string.h>
+#include <device.h>
+#include <init.h>
+#include <drivers/display.h>
+#include <drivers/gpio.h>
+#include <drivers/spi.h>
+#include <sys/byteorder.h>
+
+#include "ek79652_regs.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ek79652, CONFIG_DISPLAY_LOG_LEVEL);
+
+/**
+ * EK79652 compatible EPD controller driver.
+ *
+ */
+#define EK79652_DC_PIN			DT_INST_GPIO_PIN(0, dc_gpios)
+#define EK79652_DC_FLAGS		DT_INST_GPIO_FLAGS(0, dc_gpios)
+#define EK79652_DC_CNTRL		DT_INST_GPIO_LABEL(0, dc_gpios)
+#define EK79652_BUSY_PIN		DT_INST_GPIO_PIN(0, busy_gpios)
+#define EK79652_BUSY_CNTRL		DT_INST_GPIO_LABEL(0, busy_gpios)
+#define EK79652_BUSY_FLAGS		DT_INST_GPIO_FLAGS(0, busy_gpios)
+#define EK79652_RESET_PIN		DT_INST_GPIO_PIN(0, reset_gpios)
+#define EK79652_RESET_CNTRL		DT_INST_GPIO_LABEL(0, reset_gpios)
+#define EK79652_RESET_FLAGS		DT_INST_GPIO_FLAGS(0, reset_gpios)
+
+#define EPD_PANEL_WIDTH			DT_INST_PROP(0, width)
+#define EPD_PANEL_HEIGHT		DT_INST_PROP(0, height)
+#define EK79652_PIXELS_PER_BYTE		8U
+
+/* Horizontally aligned page! */
+#define EK79652_NUMOF_PAGES		(EPD_PANEL_WIDTH / \
+					 EK79652_PIXELS_PER_BYTE)
+#define EK79652_BUFFER_SIZE		(EK79652_NUMOF_PAGES * EPD_PANEL_HEIGHT)
+#define EK79652_PANEL_FIRST_GATE	0U
+#define EK79652_PANEL_LAST_GATE		(EPD_PANEL_HEIGHT - 1)
+#define EK79652_PANEL_FIRST_PAGE	0U
+#define EK79652_PANEL_LAST_PAGE		(EK79652_NUMOF_PAGES - 1)
+
+struct ek79652_data {
+	const struct ek79652_config *config;
```

Comment:
That is wrong, just as it is wrong in gd7965 driver (I will fix gd7965 later), see https://github.com/zephyrproject-rtos/zephyr/pull/39708/commits/63f48e70bb5aea39533c4f7bd6fcbe6db1e0ff2a (part of https://github.com/zephyrproject-rtos/zephyr/pull/39708)